staging/src/k8s.io/apimachinery/pkg/util/managedfields/extract.go

```diff
@@ -66,10 +70,13 @@ func ExtractInto(object runtime.Object, objectType typed.ParseableType, fieldMan
 	return nil
 }
 
-func findManagedFields(accessor metav1.Object, fieldManager string) (metav1.ManagedFieldsEntry, bool) {
+func findManagedFields(accessor metav1.Object, fieldManager string, subresource string) (metav1.ManagedFieldsEntry, bool) {
 	objManagedFields := accessor.GetManagedFields()
 	for _, mf := range objManagedFields {
-		if mf.Manager == fieldManager && mf.Operation == metav1.ManagedFieldsOperationApply {
+		// TODO(jpbetz): The addition of subresource is backward incompatible with 1.21 clients. What to do?
```

Comment:
Can you elaborate on what this means?

Reply:
I'll remove this comment since it's actually not that bad of an issue.

Explanation: Logically, the addition of the "subresource" field to managed fields in https://github.com/kubernetes/kubernetes/pull/98377 is a backward incompatible change. Once the server begins sending read responses that include managed fields that use the "subresource" field, existing clients that were previously consuming the managed field data can be broken. The circumstances where this can actually happen in practices are very specific:
- field manager X applies the SPEC of object Y
- field manager X applies the STATUS of object Y
- client reads object Y and accesses the field manager data for manager X by looking for the first field manager entry that has manager name X
- client expects to get back the SPEC managed field data but instead gets back the STATUS managed field data

The reason this is not expected to happen in practice is it requires that the same field manager be used for both SPEC and STATUS, which is not strongly discouraged.